webapp/src/ts/modals/guided-setup/guided-setup.component.ts

```diff
@@ -0,0 +1,171 @@
+import { COUNTRY_LIST } from '../../providers/countries.provider';
+
+@Component({
+  selector: 'guided-setup',
+  templateUrl: './guided-setup.component.html'
+})
+export class GuidedSetupComponent extends MmModalAbstract implements AfterViewInit, AfterContentChecked {
+  private globalactions;
+  model:{ countryCode?, gatewayNumber? } = {};
+  error:{ message? } = {};
+  enabledLocales;
+  countryList;
+  settingsLoaded = false;
+  settings = <any>{};
+
+  constructor(
+    public bsModalRef: BsModalRef,
+    private store: Store,
+    private updateSettingsService: UpdateSettingsService,
+    private langugesService: LanguagesService,
+    private settingsService: SettingsService,
+  ) {
+    super();
+    this.globalactions = new GlobalActions(store);
+  }
+
+  private updateNumbers() {
+    const gatewayNumber = $('#guided-setup [name=gateway-number]').val();
+    const defaultCountryCode = $('#guided-setup [name=default-country-code]').val();
+    const parts = [];
+    if (defaultCountryCode) {
+      parts.push('+' + defaultCountryCode);
+    }
+    if (gatewayNumber) {
+      parts.push(gatewayNumber);
+    }
+    $(this).closest('.panel').find('.panel-heading .value').text(parts.join(', '));
+    if (gatewayNumber && defaultCountryCode) {
+      $(this).closest('.panel').addClass('panel-complete');
+    }
+  };
+
+  private selectOption(e) {
+    e.preventDefault();
+    const elem = $(e.currentTarget);
+    elem.closest('.horizontal-options')
+      .find('.selected')
+      .removeClass('selected');
+    elem.addClass('selected');
+    const panel = elem.closest('.panel');
+    const label = [];
+    panel.find('.horizontal-options .selected').each(function() {
+      label.push($(this).text().trim());
+    });
+    panel
+      .addClass('panel-complete')
+      .find('.panel-heading .value')
+      .text(label.join(', '));
+  };
+
+  ngAfterViewInit() {
+    this.bsModalRef.setClass('modal-lg');
+    this.langugesService.get().then(languages => {
+      this.enabledLocales = languages;
+      this.countryList = COUNTRY_LIST;
+      $('#guided-setup').on('click', '.horizontal-options a', this.selectOption);
+      $('#guided-setup [name=gateway-number]').on('input', this.updateNumbers);
+      (<any>$('#guided-setup [name=default-country-code]')).select2({ width: '20em', data: this.countryList });
+      $('#guided-setup [name=default-country-code]').on('change', this.updateNumbers);
+      this.settingsService.get().then((res: any) => {
+        this.settingsLoaded = true;
+        this.settings = res;
+        if (res.setup_complete) {
+          $('#guided-setup [name=gateway-number]').val(this.settings.gateway_number).trigger('input');
+          $('#primary-contact-content a[data-value=' + this.settings.care_coordinator + ']').trigger('click');
+          $('#registration-form-content a[data-value=' + this.settings.anc_registration_lmp + ']').trigger('click');
+        }
+      })
+    });
+  }
+
+  ngAfterContentChecked() {
```

Comment:
When I implemented the other 2 comments, I noticed that this runs _before_ the view is updated. 
I got the expected result with using `ngAfterViewChecked` instead of `ngAfterContentChecked`.